agents/hermes/runtime-config-guard.py

```
@@ -352,6 +349,108 @@ def _normalize_provider_placeholder_for_env_key(value: str, env_key: str) -> str
+
+
+def _validate_env_text_with_boundary(text: str, boundary_validator_path: str | None) -> None:
+    if not boundary_validator_path:
+        raise UnsafePathError("Hermes provider placeholder refresh requires the secret-boundary validator")
+    fd, temp_path = tempfile.mkstemp(prefix="hermes-env-boundary-", text=True)
+    try:
+        with os.fdopen(fd, "w", encoding="utf-8") as handle:
+            fd = -1
+            handle.write(text)
+        try:
+            result = subprocess.run(
```

Ruff (0.15.18)

[error] `subprocess` call: check for execution of untrusted input

(S603)